src/views/SelectProduct.vue

```diff
@@ -250,16 +234,58 @@ export default defineComponent({
             "q.op": "AND"
           } as any,
           "query": "*:*",
-          "filter": "docType: PRODUCT"
+          "filter": ["docType: PRODUCT"]
         }
+      } as any
+    }
+  },
+  methods: {
+    updateInclusionQuery(value: string, type: string) {
+      this.included[type].includes(value) ? this.included[type].splice(this.included[type].indexOf(value), 1) : this.included[type].push(value)
+      this.updateQuery();
+    },
+    updateExclusionQuery(value: string, type: string) {
+      this.excluded[type].includes(value) ? this.excluded[type].splice(this.excluded[type].indexOf(value), 1) : this.excluded[type].push(value)
+      this.updateQuery();
+    },
+    updateQuery() {
+      // initializing the filter always on updateQuery call because we are adding values in the filter
+      // as string and if some value is removed then we need to do multiple operations on the filter string
+      // to remove that value from the query filter
+      this.query.json['filter'] = ["docType: PRODUCT"]
+
+      this.included['tags'].length > 0 && this.query.json['filter'].push(`tags: (${this.included['tags'].join(' OR ')})`)
```

Comment:
We could iterate using reduce and prepare it dynamically

Reply:
Sir, used reduce to prepare the query.